### PR TITLE
Feature/void new storage

### PR DESCRIPTION
**Pull request by benoitdevos**
This PR is an implementation of the `make_void()` for the pallet `logionLoc`. It uses an extra storage in order to stay backward-compatible with the existing chain data.

### CODE REVIEW

File: pallets/logion_loc/src/lib.rs [hunk 1]
```diff
@@ -284,6 +290,50 @@ pub mod pallet {
...
+				if !<LocMap<T>>::contains_key(&replacer) {
+					Err(Error::<T>::ReplacerLocNotFound)?
+				} else {
+					let replacer_loc = <LocMap<T>>::get(&replacer).unwrap();
+					if replacer_loc.owner != who {
+						Err(Error::<T>::Unauthorized)?
+					}
+				}
+			}
+			let replacer = replacer_loc_id.map(|id| LocLink { id, nature: "replacer".as_bytes().to_vec() });
```
Comment (gdethier) [inline pallets/logion_loc/src/lib.rs:324]: As discussed, I am not really a fan of hard coding an artificial nature here. Let's ask Elie what he thinks about this (does it make sense to add a nature to the link?).

File: pallets/logion_loc/src/lib.rs [hunk 2]
```diff
@@ -314,4 +363,17 @@ pub mod pallet {
...
+		V1 {
+			reason: Vec<u8>,
+			replacer: Option<LocLink<LocId>>,
+		}
+	}
+
+	/// All LOC void infos, indexed by LOC ID.
+	#[pallet::storage]
+	#[pallet::getter(fn loc_void_info)]
+	pub type LocVoidInfoMap<T> = StorageMap<_, Blake2_128Concat, <T as Config>::LocId, LocVoidInfo<<T as Config>::LocId>>;
```
Comment (gdethier) [inline pallets/logion_loc/src/lib.rs:378]: Maybe move this storage item with the others.

**Review by gdethier — APPROVED** [review]
Nice fallback ;-)

File: pallets/logion_loc/src/lib.rs (hunk 1 above)
Comment (benoitdevos) [reply pallets/logion_loc/src/lib.rs:324]: > As discussed, I am not really a fan of hard coding an artificial nature here. Let's ask Elie what he thinks about this (does it make sense to add a nature to the link?).

The LocLink and thus the nature was removed in [this PR](https://github.com/logion-network/logion-node/pull/20).

**Comment by benoitdevos** [comment]
This PR is a fallback approach, aimed at extending storage rather than migrating.
Migration is the chosen solution, see [this PR](https://github.com/logion-network/logion-node/pull/20)

**Pull request closed without merging**